include/exec/task.hpp

```diff
@@ -406,10 +389,10 @@ namespace exec {
         using __context_t = typename _Context::template promise_context_t<__promise>;
 
         auto get_env() const noexcept -> const __context_t& {
-          return __context_;
+          return *__context_;
         }
 
-        __context_t __context_;
+        std::optional<__context_t> __context_;
```

Comment:
pls replace this use of `std::optional` with `__optional` from `__detail/__optional.hpp`. it is easier on the compiler.